src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java

```diff
@@ -220,37 +218,12 @@ public void verifyAgencyAccess(final String agencyId, boolean allowInactive) {
         if (AuthenticationFacade.Companion.hasRoles("INACTIVE_BOOKINGS")) {
             agencyIds.addAll(Set.of("OUT", "TRN"));
         }
-        if (agencyIds.isEmpty()) {
```

Comment:
This is the main change - this now follows the same pattern as BookingService - just throwing a single exception when needed and logging removed